fail2ban/server/filter.py

```diff
@@ -898,22 +898,18 @@ def textToIp(text, useDns):
 		return ipList
 
 	@staticmethod
-	def cidr(i, n):
-		""" Convert an IP address string with a CIDR mask into a 32-bit
-			integer.
+	def addr2bin(ipstring, cidr=None):
+		""" Convert a string IPv4 address into decimal form.
+		If cidr is supplied, return the network address for the given block
 		"""
-		# 32-bit IPv4 address mask
-		MASK = 0xFFFFFFFFL
-		return ~(MASK >> n) & MASK & DNSUtils.addr2bin(i)
-
-	@staticmethod
-	def addr2bin(string):
-		""" Convert a string IPv4 address into an unsigned integer.
-		"""
-		return struct.unpack("!L", socket.inet_aton(string))[0]
+		if cidr is None:
+			return struct.unpack("!L", socket.inet_aton(ipstring))[0]
```

Comment:
'decimal' is "A number expressed in the scale of tens".   From executing it you get 4 bytes representing an integer (internally represented in scale of twos, thus binary :-P) -- it is just printed on the screen usually  by default in a decimal format and that doesn't make it a decimal value. So let's just go to your comment 2 comments back ;)